src/plugins/intel_gpu/src/plugin/transformations/unsqueeze_broadcast_reshape_matmul_fusion.cpp

```diff
@@ -25,118 +26,63 @@ UnsqueezeBroadcastReshapeMatmulFusion::UnsqueezeBroadcastReshapeMatmulFusion() {
-
-        auto valid_transpose_order = [](const std::vector<int64_t>& order) {
-            return order.size() == 4 && order[1] == 2;
-        };
-
-        if (pattern_map.count(unsqueeze_a_m) > 0) {
-            if (!valid_transpose_order(order_a))
-                return false;
-            auto unsqueeze_a = std::dynamic_pointer_cast<ov::op::v0::Unsqueeze>(pattern_map.at(unsqueeze_a_m).get_node_shared_ptr());
-            if (!unsqueeze_a)
-                return false;
-            input_a_output_idx = unsqueeze_a->get_input_source_output(0).get_index();
-        }
-        if (pattern_map.count(unsqueeze_b_m) > 0) {
-            if (!valid_transpose_order(order_b))
-                return false;
-            auto unsqueeze_b = std::dynamic_pointer_cast<ov::op::v0::Unsqueeze>(pattern_map.at(unsqueeze_b_m).get_node_shared_ptr());
-            if (!unsqueeze_b)
-                return false;
-            input_b_output_idx = unsqueeze_b->get_input_source_output(0).get_index();
-        }
-
-        auto target_shape_a = std::vector<int32_t>();
-        auto target_shape_b = std::vector<int32_t>();
+        const auto& pattern_map = m.get_pattern_value_map();
 
         auto valid_broadcast_target_shape = [](const std::vector<int32_t>& target_shape) {
             return std::count_if(target_shape.begin(), target_shape.end(), [](int32_t s) { return s != 1; }) == 1;
         };
-
-        if (pattern_map.count(broadcast_a_m) > 0) {
-            auto broadcast_a = std::dynamic_pointer_cast<ov::op::v3::Broadcast>(pattern_map.at(broadcast_a_m).get_node_shared_ptr());
-            if (!broadcast_a || broadcast_a->get_broadcast_spec().m_type != ov::op::BroadcastType::BIDIRECTIONAL)
-                return false;
-            auto broadcast_a_target_shape = std::dynamic_pointer_cast<ov::op::v0::Constant>(pattern_map.at(broadcast_a_target_shape_m).get_node_shared_ptr());
-            target_shape_a = broadcast_a_target_shape->cast_vector<int32_t>();
-            if (!valid_broadcast_target_shape(target_shape_a))
-                return false;
-        }
-        if (pattern_map.count(broadcast_b_m) > 0) {
-            auto broadcast_b = std::dynamic_pointer_cast<ov::op::v3::Broadcast>(pattern_map.at(broadcast_b_m).get_node_shared_ptr());
-            if (!broadcast_b || broadcast_b->get_broadcast_spec().m_type != ov::op::BroadcastType::BIDIRECTIONAL)
-                return false;
-            auto broadcast_b_target_shape = std::dynamic_pointer_cast<ov::op::v0::Constant>(pattern_map.at(broadcast_b_target_shape_m).get_node_shared_ptr());
-            target_shape_b = broadcast_b_target_shape->cast_vector<int32_t>();
-            if (!valid_broadcast_target_shape(target_shape_b))
+        auto broadcast = std::dynamic_pointer_cast<ov::op::v3::Broadcast>(pattern_map.at(broadcast_m).get_node_shared_ptr());
+        if (!broadcast || broadcast->get_broadcast_spec().m_type != ov::op::BroadcastType::BIDIRECTIONAL)
```

Comment:
Please move this check to broadcast predicate

Reply:
moved it